### PR TITLE
Language-specific "build an app" tutorials; GORM instructions on Go page

### PR DESCRIPTION
**Pull request by jseldess**
This PR splits the "Build a Test App" tutorial into language-specific pages. The Go page features instructions both for using CockroachDB with a driver and with the GORM ORM. Also, the new overview page calls out ORMs that will be supported soon and encourages users to post issues they have with our recommended ORMs and to suggest additional ORMs we should investigate.

---
This change is [Reviewable](https://reviewable.io/reviews/cockroachdb/docs/1069)

### CODE REVIEW

**Comment by cuongdo** [comment]
Review status: 0 of 4 files reviewed at latest revision, 6 unresolved discussions.

---

*[connect-a-go-app-to-cockroachdb.md, line 8 at r1](https://reviewable.io:443/reviews/cockroachdb/docs/1069#-Kc4kc1EUH0mbbqwRdqn:-Kc4kc1EUH0mbbqwRdqo:b-hp1qxh) ([raw file](https://github.com/cockroachdb/docs/blob/1409815f6bf9825f4a9c2e19b06e5052bf66ceb1/connect-a-go-app-to-cockroachdb.md#L8)):*
> ```Markdown
> ---
> 
> This tutorial shows you how to connect a simple Go application to a CockroachDB cluster using either a low-level client driver or an ORM. You can use any PostgreSQL-compatible client drivers or ORMs, but we've tested and can recommend the [Go pg driver](https://godoc.org/github.com/lib/pq) and [GORM ORM](http://jinzhu.me/gorm/), so those are featured here.
> ```

s/Go pg driver/Go pq driver/

---

*[connect-a-go-app-to-cockroachdb.md, line 38 at r1](https://reviewable.io:443/reviews/cockroachdb/docs/1069#-Kc4kzpZpm6ORlkDpy-m:-Kc4kzpZpm6ORlkDpy-n:b-ohk621) ([raw file](https://github.com/cockroachdb/docs/blob/1409815f6bf9825f4a9c2e19b06e5052bf66ceb1/connect-a-go-app-to-cockroachdb.md#L38)):*
> ```Markdown
> ## Step 1. Install the Client Driver
> 
> To install the [Go pg driver](https://godoc.org/github.com/lib/pq), run the following command:
> ```

s/pg/pq/

---

*[connect-a-go-app-to-cockroachdb.md, line 98 at r1](https://reviewable.io:443/reviews/cockroachdb/docs/1069#-Kc4nOLm2bT6qIT7eWh_:-Kc4nOLnuDWOrFBOi11G:b-fhgo0y) ([raw file](https://github.com/cockroachdb/docs/blob/1409815f6bf9825f4a9c2e19b06e5052bf66ceb1/connect-a-go-app-to-cockroachdb.md#L98)):*
> ```Markdown
> 
> ~~~ shell
> $ cockroach sql -e 'CREATE DATABASE bank'
> ```

is there a simple way to add a "copy to clipboard" button to example shell commands? (if not, don't worry about it)

---

*[connect-a-go-app-to-cockroachdb.md, line 188 at r1](https://reviewable.io:443/reviews/cockroachdb/docs/1069#-Kc4mET4LSqRP7bZBdyP:-Kc4mET4LSqRP7bZBdyQ:bq08b6s) ([raw file](https://github.com/cockroachdb/docs/blob/1409815f6bf9825f4a9c2e19b06e5052bf66ceb1/connect-a-go-app-to-cockroachdb.md#L188)):*
> ```Markdown
> ## Step 5. Run the Go code
> 
> The following code use the [GORM](http://jinzhu.me/gorm/) ORM to map Go-specific objects to SQL operations. Specifically, `db.AutoMigrate(&Account{})` creates an `accounts` table based on the Account model, the `db.Create()` calls insert rows into the table, and `db.Find(&accounts)` selects from the table so that balances can be printed.
> ```

to make these clauses parallel, remove the "the" before `db.Create()`

---

*[connect-a-go-app-to-cockroachdb.md, line 242 at r1](https://reviewable.io:443/reviews/cockroachdb/docs/1069#-Kc4m_TBZo7dSGbR17sd:-Kc4m_TBZo7dSGbR17se:b78py0r) ([raw file](https://github.com/cockroachdb/docs/blob/1409815f6bf9825f4a9c2e19b06e5052bf66ceb1/connect-a-go-app-to-cockroachdb.md#L242)):*
> ```Markdown
> </div>
> 
> ## What's Next?
> ```

I think there should be a link to the GORM docs here

---

*[connect-a-go-app-to-cockroachdb.md, line 253 at r1](https://reviewable.io:443/reviews/cockroachdb/docs/1069#-Kc4mrTz35PZCvaNcAIY:-Kc4mrTz35PZCvaNcAIZ:b-v4le3v) ([raw file](https://github.com/cockroachdb/docs/blob/1409815f6bf9825f4a9c2e19b06e5052bf66ceb1/connect-a-go-app-to-cockroachdb.md#L253)):*
> ```Markdown
> $(document).ready(function() {
>   // Handler for .ready() called.
>   $('#toc').toc({ minimumHeaders: 0, listType: 'ul', showSpeed: 0, headers: 'h2:not(.filter-content:not(.current) h2),h3:not(.filter-content:not(.current) h3)' });
> ```

Since this is repeated below, can you put this in its function, like this?

```
function renderTOC() {
  $('#toc').toc({ minimumHeaders: 0, listType: 'ul', showSpeed: 0, headers: 'h2:not(.filter-content:not(.current) h2),h3:not(.filter-content:not(.current) h3)' });
}
```

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/docs/1069)*

**Comment by cuongdo** [comment]
looking good overall with a few comments

---

Review status: 0 of 4 files reviewed at latest revision, 6 unresolved discussions.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/docs/1069#-:-Kc4nc23xIgE2OxM-Dgq:b-dvmpzz)*

File: _data/sidebar_doc.yml [hunk 1]
```diff
@@ -34,10 +34,36 @@ entries:
                 - title: Use the Built-in SQL Client
                   url: /use-the-built-in-sql-client.html
 
-        - title: Build a Test App
-          url: /build-a-test-app.html
+            - title: Connect an App
```
Comment (spencerkimball) [inline _data/sidebar_doc.yml]: "Connect an App" is confusing to me. Is this about building an app, or connecting from an app to CockroachDB?

How about "Build an App"?

File: connect-a-go-app-to-cockroachdb.md [hunk 2]
```diff
@@ -0,0 +1,276 @@
+---
+title: Connect a Go App to CockroachDB
+summary: Learn how to connect a simple Go application to a CockroachDB cluster using either a low-level client driver or an ORM.
```
Comment (spencerkimball) [inline connect-a-go-app-to-cockroachdb.md]: s/Learn how to connect a simple Go application to a CockroachDB cluster/Learn how to use CockroachDB from a simple Go application/

**Comment by bdarnell** [comment]
Reviewed 4 of 4 files at r1.
Review status: all files reviewed at latest revision, 10 unresolved discussions.

---

*[connect-a-go-app-to-cockroachdb.md, line 59 at r1](https://reviewable.io:443/reviews/cockroachdb/docs/1069#-Kc5SN3tVdWkeHA50dgP:-Kc5SN3tVdWkeHA50dgQ:b-js8axm) ([raw file](https://github.com/cockroachdb/docs/blob/1409815f6bf9825f4a9c2e19b06e5052bf66ceb1/connect-a-go-app-to-cockroachdb.md#L59)):*
> ```Markdown
> ## Step 2. Start a cluster
> 
> For the purpose of this tutorial, you need only one CockroachDB node running:
> ```

Mention that the sample code assumes a node running in insecure mode. We should also document how to modify the app to use secure mode (this may be a more useful variant than the version below with three local nodes). 

---

*[connect-a-go-app-to-cockroachdb.md, line 191 at r1](https://reviewable.io:443/reviews/cockroachdb/docs/1069#-Kc5SuR63fhl1Le2yEZV:-Kc5SuR63fhl1Le2yEZW:bqta51k) ([raw file](https://github.com/cockroachdb/docs/blob/1409815f6bf9825f4a9c2e19b06e5052bf66ceb1/connect-a-go-app-to-cockroachdb.md#L191)):*
> ```Markdown
> 
> Copy the code or 
> <a href="https://raw.githubusercontent.com/cockroachdb/docs/gh-pages/_includes/app/gorm-basic-sample.go" download>download it directly</a>.
> ```

Did you forget to `git add` the new file?

---

*[_data/sidebar_doc.yml, line 37 at r1](https://reviewable.io:443/reviews/cockroachdb/docs/1069#-Kc4fd_ASx-ZGeRkrtBG-r1-37:-Kc5QDhGtoPXP7cQrCc6:bfb3ynu) ([raw file](https://github.com/cockroachdb/docs/blob/1409815f6bf9825f4a9c2e19b06e5052bf66ceb1/_data/sidebar_doc.yml#L37)):*
<details><summary><i>Previously, spencerkimball (Spencer Kimball) wrote…</i></summary><blockquote>

"Connect an App" is confusing to me. Is this about building an app, or connecting from an app to CockroachDB?

How about "Build an App"?
</blockquote></details>

+1. "Connect" sounds to me like it's just about using the basic driver. I think either "build" or "use" is the verb we want here. How about "Use CockroachDB from..."?

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/docs/1069)*

**Comment by jseldess** [comment]
Review status: all files reviewed at latest revision, 10 unresolved discussions.

---

*[connect-a-go-app-to-cockroachdb.md, line 3 at r1](https://reviewable.io:443/reviews/cockroachdb/docs/1069#-Kc4fd_ASx-ZGeRkrtBL-r1-3:-Kc6ACiQxKOwBBj3__eZ:b-896fix) ([raw file](https://github.com/cockroachdb/docs/blob/1409815f6bf9825f4a9c2e19b06e5052bf66ceb1/connect-a-go-app-to-cockroachdb.md#L3)):*
<details><summary><i>Previously, spencerkimball (Spencer Kimball) wrote…</i></summary><blockquote>

s/Learn how to connect a simple Go application to a CockroachDB cluster/Learn how to use CockroachDB from a simple Go application/
</blockquote></details>

Done.

---

*[connect-a-go-app-to-cockroachdb.md, line 8 at r1](https://reviewable.io:443/reviews/cockroachdb/docs/1069#-Kc4kc1EUH0mbbqwRdqn:-Kc6AHCgPP7QIBYOmOFJ:b-896fix) ([raw file](https://github.com/cockroachdb/docs/blob/1409815f6bf9825f4a9c2e19b06e5052bf66ceb1/connect-a-go-app-to-cockroachdb.md#L8)):*
<details><summary><i>Previously, cuongdo (Cuong Do) wrote…</i></summary><blockquote>

s/Go pg driver/Go pq driver/
</blockquote></details>

Done.

---

*[connect-a-go-app-to-cockroachdb.md, line 38 at r1](https://reviewable.io:443/reviews/cockroachdb/docs/1069#-Kc4kzpZpm6ORlkDpy-m:-Kc6AJpxwxNYtyUy4oin:b-896fix) ([raw file](https://github.com/cockroachdb/docs/blob/1409815f6bf9825f4a9c2e19b06e5052bf66ceb1/connect-a-go-app-to-cockroachdb.md#L38)):*
<details><summary><i>Previously, cuongdo (Cuong Do) wrote…</i></summary><blockquote>

s/pg/pq/
</blockquote></details>

Done.

---

*[connect-a-go-app-to-cockroachdb.md, line 59 at r1](https://reviewable.io:443/reviews/cockroachdb/docs/1069#-Kc5SN3tVdWkeHA50dgP:-Kc6APYOyR8wxUXqBXcE:byrm97g) ([raw file](https://github.com/cockroachdb/docs/blob/1409815f6bf9825f4a9c2e19b06e5052bf66ceb1/connect-a-go-app-to-cockroachdb.md#L59)):*
<details><summary><i>Previously, bdarnell (Ben Darnell) wrote…</i></summary><blockquote>

Mention that the sample code assumes a node running in insecure mode. We should also document how to modify the app to use secure mode (this may be a more useful variant than the version below with three local nodes). 
</blockquote></details>

Added "insecure mode". 

But I  have another issue to adjust the code sample for a secure cluster: https://github.com/cockroachdb/docs/issues/631. I'll try to get to that soon, and when I do, I'll update this `cockroach start` step as well. 

---

*[connect-a-go-app-to-cockroachdb.md, line 98 at r1](https://reviewable.io:443/reviews/cockroachdb/docs/1069#-Kc4nOLm2bT6qIT7eWh_:-Kc6AzOMdNA_BUDi5fQq:b-acqpni) ([raw file](https://github.com/cockroachdb/docs/blob/1409815f6bf9825f4a9c2e19b06e5052bf66ceb1/connect-a-go-app-to-cockroachdb.md#L98)):*
<details><summary><i>Previously, cuongdo (Cuong Do) wrote…</i></summary><blockquote>

is there a simple way to add a "copy to clipboard" button to example shell commands? (if not, don't worry about it)
</blockquote></details>

I'd rather handle that separately, since it's functionality that will apply to and benefit all other pages with shell and code samples. I created an issue: https://github.com/cockroachdb/docs/issues/1070

---

*[connect-a-go-app-to-cockroachdb.md, line 188 at r1](https://reviewable.io:443/reviews/cockroachdb/docs/1069#-Kc4mET4LSqRP7bZBdyP:-Kc6Bdf0V9gUA2QnLZ8q:b-896fix) ([raw file](https://github.com/cockroachdb/docs/blob/1409815f6bf9825f4a9c2e19b06e5052bf66ceb1/connect-a-go-app-to-cockroachdb.md#L188)):*
<details><summary><i>Previously, cuongdo (Cuong Do) wrote…</i></summary><blockquote>

to make these clauses parallel, remove the "the" before `db.Create()`
</blockquote></details>

Done.

---

*[connect-a-go-app-to-cockroachdb.md, line 191 at r1](https://reviewable.io:443/reviews/cockroachdb/docs/1069#-Kc5SuR63fhl1Le2yEZV:-Kc6BnGVGHo7IyBGmkbP:b-ugo68r) ([raw file](https://github.com/cockroachdb/docs/blob/1409815f6bf9825f4a9c2e19b06e5052bf66ceb1/connect-a-go-app-to-cockroachdb.md#L191)):*
<details><summary><i>Previously, bdarnell (Ben Darnell) wrote…</i></summary><blockquote>

Did you forget to `git add` the new file?
</blockquote></details>

Cuong added the file in a separate PR: https://github.com/cockroachdb/docs/pull/1054. Don't you see it? 

---

*[connect-a-go-app-to-cockroachdb.md, line 242 at r1](https://reviewable.io:443/reviews/cockroachdb/docs/1069#-Kc4m_TBZo7dSGbR17sd:-Kc6FZabk7d93gAppTKt:b-896fix) ([raw file](https://github.com/cockroachdb/docs/blob/1409815f6bf9825f4a9c2e19b06e5052bf66ceb1/connect-a-go-app-to-cockroachdb.md#L242)):*
<details><summary><i>Previously, cuongdo (Cuong Do) wrote…</i></summary><blockquote>

I think there should be a link to the GORM docs here
</blockquote></details>

Done.

---

*[connect-a-go-app-to-cockroachdb.md, line 253 at r1](https://reviewable.io:443/reviews/cockroachdb/docs/1069#-Kc4mrTz35PZCvaNcAIY:-Kc6IPujTeW5RJATS4QR:bd29sez) ([raw file](https://github.com/cockroachdb/docs/blob/1409815f6bf9825f4a9c2e19b06e5052bf66ceb1/connect-a-go-app-to-cockroachdb.md#L253)):*
<details><summary><i>Previously, cuongdo (Cuong Do) wrote…</i></summary><blockquote>

Since this is repeated below, can you put this in its function, like this?

```
function renderTOC() {
  $('#toc').toc({ minimumHeaders: 0, listType: 'ul', showSpeed: 0, headers: 'h2:not(.filter-content:not(.current) h2),h3:not(.filter-content:not(.current) h3)' });
}
```
</blockquote></details>

Done. Look ok?

---

*[_data/sidebar_doc.yml, line 37 at r1](https://reviewable.io:443/reviews/cockroachdb/docs/1069#-Kc4fd_ASx-ZGeRkrtBG-r1-37:-Kc6Oox48G4vkmYbddRl:b84hikn) ([raw file](https://github.com/cockroachdb/docs/blob/1409815f6bf9825f4a9c2e19b06e5052bf66ceb1/_data/sidebar_doc.yml#L37)):*
<details><summary><i>Previously, bdarnell (Ben Darnell) wrote…</i></summary><blockquote>

+1. "Connect" sounds to me like it's just about using the basic driver. I think either "build" or "use" is the verb we want here. How about "Use CockroachDB from..."?
</blockquote></details>

Thanks for this feedback, @spencerkimball and @bdarnell. This was originally "Build a Test App". I'll revert to that and retitle these pages to "Build a Test App in <language>".

Though I wonder if it would be better to leave "CockroachDB" in the page titles for optimal SEO... Does this resonate? "Build a <language> App on CockroachDB"?

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/docs/1069)*

**Comment by jseldess** [comment]
*[_data/sidebar_doc.yml, line 37 at r1](https://reviewable.io:443/reviews/cockroachdb/docs/1069#-Kc4fd_ASx-ZGeRkrtBG-r1-37:-Kc6TJuvu62WkFLRgMNy:b-9k50h8) ([raw file](https://github.com/cockroachdb/docs/blob/1409815f6bf9825f4a9c2e19b06e5052bf66ceb1/_data/sidebar_doc.yml#L37)):*
<details><summary><i>Previously, jseldess wrote…</i></summary><blockquote>

Thanks for this feedback, @spencerkimball and @bdarnell. This was originally "Build a Test App". I'll revert to that and retitle these pages to "Build a Test App in <language>".

Though I wonder if it would be better to leave "CockroachDB" in the page titles for optimal SEO... Does this resonate? "Build a <language> App on CockroachDB"?
</blockquote></details>

That last suggestion was supposed to be `Build a <language> App on CockroachDB`

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/docs/1069)*

**Comment by bdarnell** [comment]
LGTM

---

Review status: 2 of 4 files reviewed at latest revision, 9 unresolved discussions.

---

*[build-a-test-app-in-go.md, line 191 at r1](https://reviewable.io:443/reviews/cockroachdb/docs/1069#-Kc5SuR63fhl1Le2yEZV:-Kc9Pz5mghn8LqYEd4YA:b-o6zx95) ([raw file](https://github.com/cockroachdb/docs/blob/1409815f6bf9825f4a9c2e19b06e5052bf66ceb1/build-a-test-app-in-go.md#L191)):*
<details><summary><i>Previously, jseldess wrote…</i></summary><blockquote>

Cuong added the file in a separate PR: https://github.com/cockroachdb/docs/pull/1054. Don't you see it? 
</blockquote></details>

OK, I wasn't looking outside of this PR. 

---

*[_data/sidebar_doc.yml, line 37 at r1](https://reviewable.io:443/reviews/cockroachdb/docs/1069#-Kc4fd_ASx-ZGeRkrtBG-r1-37:-Kc9Q9NlGQl-BoQ6d2nK:bwadq1y) ([raw file](https://github.com/cockroachdb/docs/blob/1409815f6bf9825f4a9c2e19b06e5052bf66ceb1/_data/sidebar_doc.yml#L37)):*
<details><summary><i>Previously, jseldess wrote…</i></summary><blockquote>

That last suggestion was supposed to be `Build a <language> App on CockroachDB`
</blockquote></details>

"Build a [language] app on CockroachDB" sounds good for the page titles but may be a bit wordy for the sidebar. I might say "with CockroachDB" instead of "on CockroachDB". 

I like getting rid of the word "test" here - these docs are about how you'll build your real app, and the test/sample apps are just one step to getting there. 

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/docs/1069#-:-Kc9R4tMKoF9VfXuIHRq:bb6ywmx)*

**Comment by jseldess** [comment]
Review status: 2 of 4 files reviewed at latest revision, 8 unresolved discussions.

---

*[_data/sidebar_doc.yml, line 37 at r1](https://reviewable.io:443/reviews/cockroachdb/docs/1069#-Kc4fd_ASx-ZGeRkrtBG-r1-37:-KcB7busuevrnrtwSWRW:b-896fix) ([raw file](https://github.com/cockroachdb/docs/blob/1409815f6bf9825f4a9c2e19b06e5052bf66ceb1/_data/sidebar_doc.yml#L37)):*
<details><summary><i>Previously, bdarnell (Ben Darnell) wrote…</i></summary><blockquote>

"Build a [language] app on CockroachDB" sounds good for the page titles but may be a bit wordy for the sidebar. I might say "with CockroachDB" instead of "on CockroachDB". 

I like getting rid of the word "test" here - these docs are about how you'll build your real app, and the test/sample apps are just one step to getting there. 
</blockquote></details>

Done.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/docs/1069)*

**Comment by nvanbenschoten** [comment]
:lgtm:

---

Reviewed 2 of 5 files at r1, 1 of 3 files at r2.
Review status: 3 of 4 files reviewed at latest revision, 10 unresolved discussions.

---

*[build-a-go-app-with-cockroachdb.md, line 21 at r2](https://reviewable.io:443/reviews/cockroachdb/docs/1069#-KcHvCGJqM8Y-VaxQmRM:-KcHvCGJqM8Y-VaxQmRN:b-eg8m) ([raw file](https://github.com/cockroachdb/docs/blob/b8e37058e9415aec4a9fbf458b70a4a9e2c9b798/build-a-go-app-with-cockroachdb.md#L21)):*
> ```Markdown
> </style>
> 
> <div id="tool-filters" class="filters clearfix">
> ```

I'm not exactly sure how these filters work, but is it feasible to put the selection in the URL so that refreshing the page doesn't go back to the first option? This probably applies elsewhere on our docs as well.

---

*[build-a-go-app-with-cockroachdb.md, line 183 at r2](https://reviewable.io:443/reviews/cockroachdb/docs/1069#-KcHtCOK2JTRLNTauK-l:-KcHtCOK2JTRLNTauK-m:bynv9dg) ([raw file](https://github.com/cockroachdb/docs/blob/b8e37058e9415aec4a9fbf458b70a4a9e2c9b798/build-a-go-app-with-cockroachdb.md#L183)):*
> ```Markdown
> ## Step 5. Run the Go code
> 
> The following code use the [GORM](http://jinzhu.me/gorm/) ORM to map Go-specific objects to SQL operations. Specifically, `db.AutoMigrate(&Account{})` creates an `accounts` table based on the Account model, `db.Create()` inserts rows into the table, and `db.Find(&accounts)` selects from the table so that balances can be printed.
> ```

We should go with `db.Create(&Account{})` or `db.Create(account)`. The method decides which table to insert rows into based on the object passed to it. 

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/docs/1069#-:-KcHve9Qu-oR8TQjEOi7:bnfp4nl)*

**Comment by jseldess** [comment]
Review status: 3 of 4 files reviewed at latest revision, 10 unresolved discussions.

---

*[build-a-go-app-with-cockroachdb.md, line 21 at r2](https://reviewable.io:443/reviews/cockroachdb/docs/1069#-KcHvCGJqM8Y-VaxQmRM:-KcJ5-RBxMOgWksOfQcX:b-vy6jdu) ([raw file](https://github.com/cockroachdb/docs/blob/b8e37058e9415aec4a9fbf458b70a4a9e2c9b798/build-a-go-app-with-cockroachdb.md#L21)):*
<details><summary><i>Previously, nvanbenschoten (Nathan VanBenschoten) wrote…</i></summary><blockquote>

I'm not exactly sure how these filters work, but is it feasible to put the selection in the URL so that refreshing the page doesn't go back to the first option? This probably applies elsewhere on our docs as well.
</blockquote></details>

I'd have to make two distinct pages, one with the driver instructions, one with the ORM instructions. Given that we already have one page per language, I think that'd be overkill. Though I agree that it'd be ideal for the page not to go back to the first option on refresh. 

I'll look into this a bit more, but I suspect that it's best to put tracking in place to see when readers are filtering for ORM instructions and leave the rest as is for now.

---

*[build-a-go-app-with-cockroachdb.md, line 183 at r2](https://reviewable.io:443/reviews/cockroachdb/docs/1069#-KcHtCOK2JTRLNTauK-l:-KcJ4xDohzdPjrAMkD7z:b-896fix) ([raw file](https://github.com/cockroachdb/docs/blob/b8e37058e9415aec4a9fbf458b70a4a9e2c9b798/build-a-go-app-with-cockroachdb.md#L183)):*
<details><summary><i>Previously, nvanbenschoten (Nathan VanBenschoten) wrote…</i></summary><blockquote>

We should go with `db.Create(&Account{})` or `db.Create(account)`. The method decides which table to insert rows into based on the object passed to it. 
</blockquote></details>

Done.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/docs/1069)*

**Comment by jordanlewis** [comment]
:lgtm:

---

Review status: 3 of 4 files reviewed at latest revision, 12 unresolved discussions.

---

*[build-a-go-app-with-cockroachdb.md, line 26 at r3](https://reviewable.io:443/reviews/cockroachdb/docs/1069#-KcJB6VMK0C3TJPPWBVd:-KcJB6VMK0C3TJPPWBVe:b-km1yxw) ([raw file](https://github.com/cockroachdb/docs/blob/48feb5ccd6e903aea87bf60596e551d7a4939780/build-a-go-app-with-cockroachdb.md#L26)):*
> ```Markdown
> </div>
> 
> {{site.data.alerts.callout_success}}For tutorials in other languages, see the menu to the left.{{site.data.alerts.end}}
> ```

I'm not sure if this paragraph is necessary, or if it is, whether it belongs where it is at the moment. An entire screen is currently taken up with the switcher button, this callout, and the TOC. I think it could be more ergonomic if, as you use the switcher, you can see the content changing on the same screen - just so readers understand that something is happening below The site layout is already pretty clear to me, since you can see the languages already to the left of the article, so I'm not sure the callout itself adds that much value either.

I don't have super strong feelings on this either way though, just wanted to throw it out there as a question.

---

*[build-a-go-app-with-cockroachdb.md, line 183 at r3](https://reviewable.io:443/reviews/cockroachdb/docs/1069#-KcJAcNMMbCkjMBQjv7w:-KcJAcNMMbCkjMBQjv7x:b-kjrh7n) ([raw file](https://github.com/cockroachdb/docs/blob/48feb5ccd6e903aea87bf60596e551d7a4939780/build-a-go-app-with-cockroachdb.md#L183)):*
> ```Markdown
> ## Step 5. Run the Go code
> 
> The following code use the [GORM](http://jinzhu.me/gorm/) ORM to map Go-specific objects to SQL operations. Specifically, `db.AutoMigrate(&Account{})` creates an `accounts` table based on the Account model, `db.Create(&Account{})` inserts rows into the table, and `db.Find(&accounts)` selects from the table so that balances can be printed.
> ```

use -> uses

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/docs/1069#-:-KcJB3rErCoCc-67YOvv:bnfp4nl)*

**Comment by jseldess** [comment]
Review status: 3 of 4 files reviewed at latest revision, 12 unresolved discussions.

---

*[build-a-go-app-with-cockroachdb.md, line 26 at r3](https://reviewable.io:443/reviews/cockroachdb/docs/1069#-KcJB6VMK0C3TJPPWBVd:-KcJJm4lFslvFKbS-IcC:b-h3wjd) ([raw file](https://github.com/cockroachdb/docs/blob/48feb5ccd6e903aea87bf60596e551d7a4939780/build-a-go-app-with-cockroachdb.md#L26)):*
<details><summary><i>Previously, jordanlewis (Jordan Lewis) wrote…</i></summary><blockquote>

I'm not sure if this paragraph is necessary, or if it is, whether it belongs where it is at the moment. An entire screen is currently taken up with the switcher button, this callout, and the TOC. I think it could be more ergonomic if, as you use the switcher, you can see the content changing on the same screen - just so readers understand that something is happening below The site layout is already pretty clear to me, since you can see the languages already to the left of the article, so I'm not sure the callout itself adds that much value either.

I don't have super strong feelings on this either way though, just wanted to throw it out there as a question.
</blockquote></details>

Good point, @jordanlewis. That callout really isn't necessary, so I've removed it. I also moved the Driver/ORM toggles to the very top.

---

*[build-a-go-app-with-cockroachdb.md, line 183 at r3](https://reviewable.io:443/reviews/cockroachdb/docs/1069#-KcJAcNMMbCkjMBQjv7w:-KcJK1wa5yLMDr0UqAU9:b-896fix) ([raw file](https://github.com/cockroachdb/docs/blob/48feb5ccd6e903aea87bf60596e551d7a4939780/build-a-go-app-with-cockroachdb.md#L183)):*
<details><summary><i>Previously, jordanlewis (Jordan Lewis) wrote…</i></summary><blockquote>

use -> uses
</blockquote></details>

Done.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/docs/1069)*

**Comment by jordanlewis** [comment]
Reviewed 2 of 5 files at r1, 2 of 3 files at r2, 2 of 2 files at r4.
Review status: all files reviewed at latest revision, 10 unresolved discussions.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/docs/1069)*

**Comment by nvanbenschoten** [comment]
Review status: all files reviewed at latest revision, 9 unresolved discussions.

---

*[build-a-go-app-with-cockroachdb.md, line 21 at r2](https://reviewable.io:443/reviews/cockroachdb/docs/1069#-KcHvCGJqM8Y-VaxQmRM:-KcJurHzOyerD-O1mnoW:b-554oi1) ([raw file](https://github.com/cockroachdb/docs/blob/b8e37058e9415aec4a9fbf458b70a4a9e2c9b798/build-a-go-app-with-cockroachdb.md#L21)):*
<details><summary><i>Previously, jseldess wrote…</i></summary><blockquote>

I'd have to make two distinct pages, one with the driver instructions, one with the ORM instructions. Given that we already have one page per language, I think that'd be overkill. Though I agree that it'd be ideal for the page not to go back to the first option on refresh. 

I'll look into this a bit more, but I suspect that it's best to put tracking in place to see when readers are filtering for ORM instructions and leave the rest as is for now.

</blockquote></details>

If it's difficult then I don't think it's worth trying to fix now. It's primarily just something to be aware of whenever we have a large number of these filtering options which configure the entire page.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/docs/1069)*

File: _data/sidebar_doc.yml [hunk 3]
```diff
@@ -34,10 +34,39 @@ entries:
...
+                - title: Java
+                  url: /build-a-java-app-with-cockroachdb.html
+
+                - title: Node.js
+                  url: /build-a-nodejs-app-with-cockroachdb.html
+
+                - title: C++
+                  url: /build-a-c++-app-with-cockroachdb.html
+
+                - title: Closure
```
Comment (bdarnell) [inline _data/sidebar_doc.yml]: The language is "Clojure", not "Closure".

Comment (jseldess) [inline _data/sidebar_doc.yml]: I always make that mistake. Thanks, Ben.

**Comment by jseldess** [comment]
Merging despite test failure, which is simply due to link in the footer pointing to a page not yet on the live site.
